config/routes.rb

```diff
@@ -6,6 +6,11 @@
   devise_for :users
   root to: "bots#index"
 
+  get 'bots/data.json', to: 'bots#list_data', as: :list_data
+  get 'bots/data/:data_key', to: 'bots#get_data', as: :get_data, constraints: {:data_key => /[\w.]+/}
```

Comment:
`[\w.]` will match either of `\w` or `.` - since `.` in regex is "any character", `[\w.]` matches any character. If you want to match "word characters and periods, you want `[\w\.]`.